src/MongoDB.Bson/BsonExtensionMethods.cs

```diff
@@ -43,6 +43,15 @@ public static class BsonExtensionMethods
             BsonSerializationArgs args = default(BsonSerializationArgs)
             )
         {
+            if (args.NominalType == null)
+            {
+                args.NominalType = typeof(TNominalType);
+            }
+            else if (args.NominalType != typeof(TNominalType))
```

Comment:
Should we also consider this option `SerializeAsNominalType` before throwing exception?

Reply:
I don't think so. It's not an error for the actual type to be different from the nominal type (as long as its derived from it).

Reply:
ok